extensions/amp-izlesene/0.1/amp-izlesene.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import {getDataParamsFromAttributes} from '../../../src/dom';
+ import {addParamsToUrl} from '../../../src/url';
+ import {isLayoutSizeDefined} from '../../../src/layout';
+ import {user} from '../../../src/log';
+
+ class AmpIzlesene extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url('https://www.izlesene.com', onLayout);
+    // Host that Vimeo uses to serve poster frames needed by player.
+    this.preconnect.url('https://i1.imgiz.com', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const videoid = user().assert(
```

Comment:
Let's do the following:

1- add a `constructor` and declare this.videoId_ there.
2- do the assert and assignment of videoId in `buildCallback` instead of here.
3- create a new `getVideoIframeSrc_` that returns the correct Url with all the params appended.
4- use `getVideoIframeSrc_` here and in the preconnectCallback.

(see amp-youtube which takes a similar approach)